### DISCUSSION TITLE
添加不使用 `proxy-providers` 的 `select profile`?

### DISCUSSION THREAD

**Discussion by JohanChane** [JohanChane/clashtui#30] [Ideas]
应用场景:
-   当只有一个订阅链接时, 想控制生成的 config.yaml 的所有内容 (除 proxies 字段)。但又不想用 `proxy-provider` (当手机上没有可用节点时, 更新 `proxy-providers` 不方便)。还有这样可以取消 mihomo 对 `proxy-providers` 的更新 (当然也可以通过 update-interval 字段设置不更新, 但手机端使用该配置时要设置定时更新), 使用 clashtui 的更新。
-   有些转换 clash 配置到 singbox 的工具不支持 proxy-providers。移除 proxy-providers 后, 转换更加方便。
-   解决 proxy name 重名问题。

实现思路:
-   在合并后 config (在内存中) 中操作。
-   取出所有 proxy-providers 的 proxies。
-   替换每个 `proxy-group` 的 use: <proxy-providers> 为 proxies: <proxy names>。
-   如果 proxies 有相同的名字, 则加前缀使它们都不相同。将取出的 proxies 插入到字段 `proxies`。
-   将 config 写入到 `config.yaml`。

操作:
-   `m`: 比 `select profile` 操作多一个移除 `proxy-providers` 字段的步骤。

**Comment by JohanChane** [comment]
将 `m` 改为了切换 `No proxy provider` 选项。

**Comment by Jackhr-arch** [comment]
I think there is not need for a state, another `msgpopup` will help

**Comment by JohanChane** [comment]
为了表达更加清晰, 我继续用中文讨论。

我的移动端会使用 `proxy-providers` 来自动更新。clashtui 生成配置后, 我会上传到我的 gitlab 私人仓库。但是 PC 端不需要 proxy-providers 因为有 cron。那么要怎么方便切换这个功能?

**Reply by Jackhr-arch** [reply]
Hmm, in this case, I can impl a command line arg (something like `--no-proxy-provider`) to remove it.

**Reply by Jackhr-arch** [reply]
Done. use `cargo r --all-features -- profile update -h` to check out

**Reply by JohanChane** [reply]
你的 aio 分支, 我还没有细看过, 到时你写一个使用文档吧。然后有 main 分支 README 放一个 aio 分支的链接。完善之后, 你可以 release。

**Comment by Jackhr-arch** [comment]
I've just recovered `no_proxy_provider`, now a popup will show after `select`/`update current profile`, and it will directly modify mihomo config file rather downloaded profiles. 
I haven't test them though.

**Reply by JohanChane** [reply]
嗯, 这样设计也是可以的。
